pkg/server/backuprestoreserver.go

```diff
@@ -200,19 +192,21 @@ func (b *BackupRestoreServer) runServer(ctx context.Context, restoreOpts *brtype
 	defer handler.Stop()
 
 	// Promotes member if it is a learner
-	for {
-		select {
-		case <-ctx.Done():
-			b.logger.Info("Context cancelled. Stopping retry promoting member")
-			return ctx.Err()
-		default:
-		}
-		m := member.NewMemberControl(b.config.EtcdConnectionConfig)
-		err := m.PromoteMember(ctx)
-		if err == nil {
-			break
+	if miscellaneous.IsMultiNode(b.logger) {
```

Comment:
Should we rather use the `restoreOpts.OriginalClusterSize` here, so that we don't have to decode the `configYML` again?

Reply:
Done